handlers/meta/handlers.go

```diff
@@ -215,6 +217,58 @@ func (h *handler) resolveMediaURL(mediaID string, token string, clog *courier.Ch
 	return mediaURL, err
 }
 
+type DeletionRequestData struct {
+	Algorithm string `json:"algorithm"`
+	Expires   int64  `json:"expires"`
+	IssuedAt  int64  `json:"issued_at"`
+	UserID    string `json:"user_id"`
+}
+
+type DeleteConfirmationData struct {
+	URL              string `json:"url"`
+	ConfirmationCode string `json:"confirmation_code"`
+}
+
+// deleteContactEvents is our HTTP handler function for deleting data requests
+func (h *handler) deleteContactEvents(ctx context.Context, channel courier.Channel, w http.ResponseWriter, r *http.Request, payload *DeletionRequestData, clog *courier.ChannelLog) ([]courier.Event, error) {
+	err := h.validateSignature(r)
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, err)
+	}
+
+	urn, err := urns.New(urns.Facebook, payload.UserID)
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, errors.New("invalid facebook id"))
+	}
+
+	contact, err := h.Server().Backend().GetContact(ctx, channel, urn, nil, "", false, clog)
+	if contact == nil {
+		return nil, handlers.WriteAndLogRequestIgnored(ctx, h, channel, w, r, "ignoring request, no existing contact matched")
+	}
+
+	date := parseTimestamp(payload.IssuedAt)
+
+	events := make([]courier.Event, 0, 2)
+	data := make([]any, 0, 2)
+
+	payloadJson, _ := json.Marshal(payload)
+	sentry.CaptureMessage(fmt.Sprintf("Data Deletion Request: %s", payloadJson))
+
+	event := h.Backend().NewChannelEvent(channel, courier.EventDeleteContact, urn, clog).WithOccurredOn(date).WithExtra(map[string]string{"userID": payload.UserID})
+
+	err = h.Backend().WriteChannelEvent(ctx, event, clog)
+	if err != nil {
+		return nil, err
+	}
+
+	confirmationURL := fmt.Sprintf("https://%s/channels/events/read/%s/", h.Server().Config().Domain, event.UUID())
```

Comment:
@norkans7 @ericnewcomer need to make a decision on this URL.. is it maybe odd to have event UUID in the URL and be the confirmation code? Should we use contact UUID instead? Is there a vulnerability there allowing enumeration of contact UUIDs? My inclination is a really generic URL like `https://<domain>/public/forgetme/` with a UUID as the confirmation code that they enter.

Reply:
I find that once we submit the form, we need a page to redirect to so i think we need to keep the event read view for that.
And that approach gives us flexibility to change that behavior with changing the link we send for the confirmation

Reply:
I used the same view with a get param
https://github.com/nyaruka/rapidpro/pull/5928/files